packages/tokens-studio-for-figma/src/storage/tokensStudio/dynamicSets/externalLoader.ts

```
@@ -0,0 +1,27 @@
+import { ExternalLoadOptions } from '@tokens-studio/graph-engine';
+import { Graphql, TokenSetQuery } from '@tokens-studio/sdk';
+import { GET_TOKEN_SET_QUERY } from '../graphql';
+import { tokensStudioToToken } from '../utils';
+
+export const externalLoader = async (opts: ExternalLoadOptions) => {
+  try {
+    const data = await Graphql.exec<TokenSetQuery>(
+      Graphql.op(GET_TOKEN_SET_QUERY, {
+        urn: opts.data.urn,
+      }),
+    );
+
+    const responseTokens = data.data?.tokenSet?.tokens;
+
+    if (!responseTokens) {
+      return { tokens: [] };
+    }
+
+    const tokens = responseTokens.map((token) => tokensStudioToToken(token));
+
+    return { tokens };
+  } catch (error) {
+    console.error(error);
```

can we raise a sentry exception? given this is our service we should understand when things fail

---

also, likely notifying the user that an error has happened might be better than silently failing via the console (e.g. notifyUI)